### PR TITLE
Api Management - Versions, Revisions and Releases

### PR DESCRIPTION
**Pull request by darrelmiller**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by marstr** [comment]
We made some updates in December, and moved back to working out of `master`. `current` is just a duplicate of `master` to allow for an easier transition. You can read more about all of the changes we made here: [December 2017 Refactoring](https://github.com/Azure/azure-rest-api-specs/wiki/December-2017-Refactoring)

**Comment by darrelmiller** [comment]
@marstr  Do you need me to create a new PR against master?

**Review by fearthecowboy — APPROVED** [review]
Looks OK to me.

New APIs added -- requires ARM signoff by @ravbhatnagar

**Comment by fearthecowboy** [comment]
FYI; the two validation errors are being addressed with suppressions in a different PR #2257

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json [hunk 1]
```diff
@@ -336,6 +344,291 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/apis/{apiId}/revisions": {
```
Comment (ravbhatnagar) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json]: Is revision an ARM resource? Is is possible to get an individual revision?

**Comment by ravbhatnagar** [comment]
Looks good. Just one question whether revisions is a resource and if its possible to get an individual revision.

**Comment by darrelmiller** [comment]
@ravbhatnagar There is no .../revisions/{revisionId} operation because it would be redundant with .../apis/{apiId} as every API is also a revision.  The .../revisions operation is simply a read-only concept used to drive a UI view.  It is not a ARM Resource.

**Comment by ravbhatnagar** [comment]
Signing off!

**Comment by darrelmiller** [comment]
@fearthecowboy Could you push the big green button please?

**Pull request merged**